src/Workspaces/MSBuildTest/VisualStudioMSBuildWorkspaceTests.cs

```diff
@@ -0,0 +1,219 @@
+            workspace.TryApplyChanges(project.AddMetadataReference(mref).Solution);
+            projFileText = File.ReadAllText(projFile);
+            Assert.Contains(@"<Reference Include=""MyAssembly""", projFileText);
+            Assert.Contains(@"<HintPath>..\References\MyAssembly.dll", projFileText);
+
+            // remove reference MyAssembly.dll
+            workspace.TryApplyChanges(workspace.CurrentSolution.GetProject(project.Id).RemoveMetadataReference(mref).Solution);
+            projFileText = File.ReadAllText(projFile);
+            Assert.DoesNotContain(@"<Reference Include=""MyAssembly""", projFileText);
+            Assert.DoesNotContain(@"<HintPath>..\References\MyAssembly.dll", projFileText);
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled))]
+        [Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public async Task TestAddRemoveMetadataReference_ReferenceAssembly()
+        {
+            CreateFiles(GetMultiProjectSolutionFiles()
+                .WithFile(@"CSharpProject\CSharpProject.csproj", Resources.ProjectFiles.CSharp.WithSystemNumerics));
+
+            var csProjFile = GetSolutionFileName(@"CSharpProject\CSharpProject.csproj");
+            var csProjFileText = File.ReadAllText(csProjFile);
+            Assert.True(csProjFileText.Contains(@"<Reference Include=""System.Numerics"""));
+
+            var vbProjFile = GetSolutionFileName(@"VisualBasicProject\VisualBasicProject.vbproj");
+            var vbProjFileText = File.ReadAllText(vbProjFile);
+            Assert.False(vbProjFileText.Contains(@"System.Numerics"));
+
+            using var workspace = CreateMSBuildWorkspace();
+            var solution = await workspace.OpenSolutionAsync(GetSolutionFileName("TestSolution.sln"));
+            var csProject = solution.Projects.First(p => p.Language == LanguageNames.CSharp);
+            var vbProject = solution.Projects.First(p => p.Language == LanguageNames.VisualBasic);
+
+            var numericsMetadata = csProject.MetadataReferences.Single(m => m.Display.Contains("System.Numerics"));
+
+            // add reference to System.Xaml
+            workspace.TryApplyChanges(vbProject.AddMetadataReference(numericsMetadata).Solution);
+            var newVbProjFileText = File.ReadAllText(vbProjFile);
+            Assert.Contains(@"<Reference Include=""System.Numerics""", newVbProjFileText);
+
+            // remove reference MyAssembly.dll
+            workspace.TryApplyChanges(workspace.CurrentSolution.GetProject(vbProject.Id).RemoveMetadataReference(numericsMetadata).Solution);
+            var newVbProjFileText2 = File.ReadAllText(vbProjFile);
+            Assert.DoesNotContain(@"<Reference Include=""System.Numerics""", newVbProjFileText2);
+        }
+
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public async Task TestOpenProject_WithXaml()
```

Comment:
Can this run on .NET 5, at least with some variations?

Reply:
I've tried adding a test for this and the generated source files do not get added to the workspace. The compilation contains the following diagnostic.
> error CS5001: Program does not contain a static 'Main' method suitable for an entry point

Perhaps the WPF on .NET core requires a particular target to be run...

Reply:
This is what [documentation](https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-wpf-framework) says about migrating. Note the different outputtype, tfm, and `Microsoft.Windows.Compatibility` reference (not sure if that's needed)

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net5.0-windows</TargetFramework>
    <OutputType>WinExe</OutputType>
    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
    <UseWPF>true</UseWPF>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
    <PackageReference Include="Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers" Version="0.2.211942">
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Windows.Compatibility" Version="5.0.2" />
  </ItemGroup>
</Project>
```